store/pg_engine.go

```diff
@@ -403,47 +419,137 @@ func getCutoffVersion(releaseMode common.ReleaseMode) (semver.Version, error) {
 		return semver.Version{}, fmt.Errorf("Migration path %s has no minor version", minorPathPrefix)
 	}
 	minorVersion := versions[len(versions)-1]
-	upperMinorVersion := minorVersion
-	upperMinorVersion.Minor = upperMinorVersion.Minor + 1
 
-	patchPathPrefix := fmt.Sprintf("migration/%s/%d.%d", releaseMode, minorVersion.Major, minorVersion.Minor)
+	patchPathPrefix := fmt.Sprintf("migration/%s/%d.%d", common.ReleaseModeRelease, minorVersion.Major, minorVersion.Minor)
 	names, err = fs.Glob(migrationFS, fmt.Sprintf("%s/*.sql", patchPathPrefix))
 	if err != nil {
 		return semver.Version{}, err
 	}
-	patchVersions, err := getPatchVersions(minorVersion, upperMinorVersion, semver.Version{} /* currentVersion */, names)
+	patchVersions, err := getPatchVersions(minorVersion, semver.Version{} /* currentVersion */, names)
 	if err != nil {
 		return semver.Version{}, err
 	}
-	// This should only happen in the dev environment
 	if len(patchVersions) == 0 {
-		if releaseMode == common.ReleaseModeDev {
-			return minorVersion, nil
-		}
 		return semver.Version{}, fmt.Errorf("Migration path %s has no patch version", patchPathPrefix)
 	}
 	return patchVersions[len(patchVersions)-1].version, nil
 }
 
+func migrateDev(ctx context.Context, d dbdriver.Driver, serverVersion, databaseName string, cutoffSchemaVersion semver.Version, histories []*dbdriver.MigrationHistory, l *zap.Logger) error {
+	devMigrations, err := getDevMigrations()
+	if err != nil {
+		return err
+	}
+
+	var migrations []devMigration
+	// Skip migrations that are already applied, otherwise the migration reattempt will most likely to fail with already exists error.
+	for _, m := range devMigrations {
+		if migrationExists(m.statement, histories) {
+			l.Info(fmt.Sprintf("Skip migrating dev migration file %s that's already migrated.", m.filename))
+		} else {
+			migrations = append(migrations, m)
+		}
+	}
+	if len(migrations) == 0 {
+		l.Info("Skip dev mode migration; no new version.")
+		return nil
+	}
+
+	for _, m := range migrations {
+		l.Info(fmt.Sprintf("Migrating dev %s...", m.filename))
+		// We don't use semantic versioning for dev migrations because they are not versioned yet.
```

Comment:
Need to update this comment

Reply:
I would also add a comment at https://sourcegraph.com/github.com/bytebase/bytebase/-/blob/store/pg_engine.go?L181 to mention we expect our own migration history to use semantic versions.

And here we add the comment to reference that method, thus further explain why we need to use semantic version here.

Reply:
Done.